.gitlab-ci.yml

```diff
@@ -0,0 +1,167 @@
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - linux-docker
+
+.docker-benches-env:               &docker-benches-env
+  image:                           "${CI_IMAGE}"
+  before_script:
+    - *rust-info-script
+    - ./scripts/ci/pre_cache.sh
+    - sccache -s
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - linux-docker-benches
+
+.kubernetes-env:                   &kubernetes-env
+  image:                           "${CI_IMAGE}"
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+
+.collect-artifacts:                &collect-artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     28 days
+    paths:
+      - ./artifacts/
+    reports:
+      dotenv: runner.env
+#### stage:                        lint
+
+fmt:
+  stage:                           lint
+  <<:                              *kubernetes-env
+  <<:                              *common-refs
+  script:
+    # FIXME: remove after stable fmt added to ci-linux
+    - rustup component add rustfmt
+    - cargo fmt --all -- --check
+
+clippy:
+  stage:                           lint
+  <<:                              *docker-env
+  <<:                              *common-refs
+  script:
+    # FIXME: remove after stable clippy added to ci-linux
+    - rustup component add clippy
+    - cargo clippy --all-targets
+
+check-rustdoc-links:
+  stage:                           lint
+  <<:                              *docker-env
+  <<:                              *common-refs
+  script:
+    - RUSTDOCFLAGS="--deny broken_intra_doc_links" cargo doc --workspace --no-deps --document-private-items
+
+#### stage:                       test
+
+check-code:
+  stage:                           test
+  <<:                              *docker-env
+  <<:                              *common-refs
+  script:
+    - cargo install cargo-hack
+    - cargo hack check --workspace --each-feature --all-targets
+
+test:
+  stage:                           test
+  <<:                              *docker-env
+  <<:                              *common-refs
+  script:
+    - cargo test
+
+benchmarks:
+  stage:                           benchmark
+  <<:                              *docker-benches-env
+  <<:                              *collect-artifacts
+  <<:                              *schedule-refs
+  script:
+    - cargo bench -p jsonrpsee-benchmarks -- --output-format bencher | tee output.txt
+    - mkdir artifacts
+    - cp output.txt artifacts/
+    - echo ${CI_RUNNER_DESCRIPTION}
+    - echo "RUNNER_NAME=${CI_RUNNER_DESCRIPTION}" > runner.env
+
+publish-bench:
```

Comment:
do you need to use `paritytech/ci-linux` here? It's better off with a lighter `paritytech/tools`

Reply:
>  It's better off with a lighter paritytech/tools

Agree, fixed

Reply:
then it maybe makes sense to remove that default image from k8s anchor.

Reply:
I'd prefer to keep it anchor as it is so it is equal through all our `.gitlab-ci.yml` files in different repos